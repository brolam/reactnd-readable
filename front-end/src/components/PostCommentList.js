import React from 'react'
import OrderOptions from './OrderOptions'
import PostCommentItem from './PostCommentItem'

function PostCommentList({ post, comments }) {
  return ([
    <div key="title" className="post-comments-title">
      <span>Commentes({comments.length})</span>
      <OrderOptions />
    </div>,
    <div key="comments" className={"post-page-comments"}>
      {comments.map(comment => (
        <PostCommentItem
          key={comment.id}
          comment={comment}
          onClickEditButton={e => { }}
          onClickDeleteButton={e => { }}
        />
      ))}
    </div>,
    <div key="flat-button" className="flat-button" onClick={e => { }} >
      <a className={"add " + post.category} >Add Comment</a>
    </div>
  ])
}

export default PostCommentList