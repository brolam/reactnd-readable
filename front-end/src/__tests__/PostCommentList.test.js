import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import PostCommentList from '../components/PostCommentList'

test('render without error', () => {
  ReactDOM.render(
    <PostCommentList post={postRedux} comments={postReduxComments} />,
    document.createElement('div'));
})

test('last Snapshot', () => {
  const postCommentList = renderer.create(<PostCommentList post={postRedux} comments={postReduxComments} />);
  let tree = postCommentList.toJSON();
  expect(tree).toMatchSnapshot();
})

const postRedux = {
  category: "redux",
  id: "7xf0y6ziyjabvozdd253nd",
  title: "Udacity is the best place to learn Redux",
  timestamp: 1467166872634,
  author: "thingtwo",
  body: "Everyone says so after all.",
  deleted: false,
  voteScore: 5
}

const postReduxComments = [
  {
    id: '894tuq4ut84ut8v4t8wun89g',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1468166872634,
    body: 'Hi there! I am a COMMENT.',
    author: 'thingtwo',
    voteScore: 6,
    deleted: false,
    parentDeleted: false
  },
  {
    id: '8tu4bsun805n8un48ve89',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1469479767190,
    body: 'Comments. Are. Cool.',
    author: 'thingone',
    voteScore: -5,
    deleted: false,
    parentDeleted: false
  }
]

