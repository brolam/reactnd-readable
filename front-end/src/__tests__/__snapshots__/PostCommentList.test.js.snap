// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
Array [
  <div
    className="post-comments-title"
>
    <span>
        Commentes(
        2
        )
    </span>
    <div
        className="order-options"
    >
        <select
            onChange={[Function]}
        >
            <option
                value="voteScore"
            >
                Vote Score
            </option>
            <option
                value="publishedDate"
            >
                Published date
            </option>
        </select>
    </div>
</div>,
  <div
    className="post-page-comments"
>
    <div
        className="post-comment"
    >
        <div
            className="post-comment-author"
        >
            by 
            thingtwo
             
            a year ago
        </div>
        <div>
            Hi there! I am a COMMENT.
        </div>
        <div
            className="post-comment-footer"
        >
            <button
                className="edit-button"
                href="/"
                onClick={[Function]}
            >
                Edit
            </button>
            <button
                className="delete-button"
                onClick={[Function]}
            >
                Delete
            </button>
            <button
                className="liked"
                onClick={[Function]}
            >
                Liked 
                <span>
                    6
                </span>
            </button>
            <button
                className="not-liked"
                onClick={[Function]}
            >
                Not like
            </button>
        </div>
    </div>
    <div
        className="post-comment"
    >
        <div
            className="post-comment-author"
        >
            by 
            thingone
             
            a year ago
        </div>
        <div>
            Comments. Are. Cool.
        </div>
        <div
            className="post-comment-footer"
        >
            <button
                className="edit-button"
                href="/"
                onClick={[Function]}
            >
                Edit
            </button>
            <button
                className="delete-button"
                onClick={[Function]}
            >
                Delete
            </button>
            <button
                className="liked"
                onClick={[Function]}
            >
                Liked 
                <span>
                    -5
                </span>
            </button>
            <button
                className="not-liked"
                onClick={[Function]}
            >
                Not like
            </button>
        </div>
    </div>
</div>,
]
`;
